Author: Stefanct
Description: Add support for Macronix MX25L1605 and SST SST25VF016B with the Lenovo Thinkpad T60/X60/X60s/X60T BIOS.
  Upstream is not against supporting this use case but even if the two
  flash chip description would be added instead of modifying existing
  ones, this patch break regular flashing and so it would likely
  require to detect the use case at runtime and switch to a different
  set of flash chip descriptions.
Forwarded: not-needed
Origin: https://www.coreboot.org/index.php?title=Board:lenovo/x60/Installation&diff=prev&oldid=11158
---
diff --git a/flashchips.c b/flashchips.c
index 58dd4f30..d9af0096 100644
--- a/flashchips.c
+++ b/flashchips.c
@@ -8052,12 +8052,12 @@ const struct flashchip flashchips[] = {
 		.name		= "MX25L1605D/MX25L1608D/MX25L1673E",
 		.bustype	= BUS_SPI,
 		.manufacture_id	= MACRONIX_ID,
-		.model_id	= MACRONIX_MX25L1605,
+		.model_id	= 0x14,
 		.total_size	= 2048,
 		.page_size	= 256,
 		.feature_bits	= FEATURE_WRSR_WREN,
 		.tested		= TEST_OK_PREW,
-		.probe		= probe_spi_rdid,
+		.probe		= probe_spi_res1,
 		.probe_timing	= TIMING_ZERO,
 		.block_erasers	=
 		{
@@ -8076,8 +8076,8 @@ const struct flashchip flashchips[] = {
 			},
 		},
 		.printlock	= spi_prettyprint_status_register_bp3_srwd, /* bit6: Continuously Program (CP) mode, for 73E is quad enable */
-		.unlock		= spi_disable_blockprotect_bp3_srwd,
-		.write		= spi_chip_write_256,
+		.unlock		= spi_disable_blockprotect,
+		.write		= spi_chip_write_1,
 		.read		= spi_chip_read, /* Fast read (0x0B), dual I/O supported */
 		.voltage	= {2700, 3600},
 	},
@@ -12719,12 +12719,12 @@ const struct flashchip flashchips[] = {
 		.name		= "SST25VF016B",
 		.bustype	= BUS_SPI,
 		.manufacture_id	= SST_ID,
-		.model_id	= SST_SST25VF016B,
+		.model_id	= 0x41,
 		.total_size	= 2048,
 		.page_size	= 256,
 		.feature_bits	= FEATURE_WRSR_EITHER,
 		.tested		= TEST_OK_PREW,
-		.probe		= probe_spi_rdid,
+		.probe		= probe_spi_res2,
 		.probe_timing	= TIMING_ZERO,
 		.block_erasers	=
 		{
@@ -12747,7 +12747,7 @@ const struct flashchip flashchips[] = {
 		},
 		.printlock	= spi_prettyprint_status_register_sst25vf016,
 		.unlock		= spi_disable_blockprotect,
-		.write		= spi_aai_write,
+		.write		= spi_chip_write_1,
 		.read		= spi_chip_read,
 		.voltage	= {2700, 3600},
 	},
